### PR TITLE
Feat: Support custom schema name for state metadata

### PR DESCRIPTION
**Pull request by eakmanrq**
Allow users to customize the schema name for state metadata. If None or empty string is provided then no schema is used and therefore the default schema for the connection/user is used instead.

### CODE REVIEW

File: sqlmesh/migrations/v0001_init.py [hunk 1]
```diff
@@ -9,11 +9,14 @@
 def migrate(state_sync):  # type: ignore
     engine_adapter = state_sync.engine_adapter
     schema = state_sync.schema
-    engine_adapter.create_schema(schema)
-
-    snapshots_table = f"{schema}._snapshots"
-    environments_table = f"{schema}._environments"
-    versions_table = f"{schema}._versions"
+    snapshots_table = f"_snapshots"
+    environments_table = f"._environments"
```
Comment (izeigerman) [inline sqlmesh/migrations/v0001_init.py:13]: `f` is no longer required here. Also how come does this work with trailing `.`?

File: sqlmesh/migrations/v0003_move_batch_size.py [hunk 2]
```diff
@@ -6,7 +6,9 @@
 def migrate(state_sync):  # type: ignore
     """Move batch_size from the model and into the kind."""
 
-    snapshots_table = f"{state_sync.schema}._snapshots"
+    snapshots_table = f"_snapshots"
```
Comment (izeigerman) [inline sqlmesh/migrations/v0003_move_batch_size.py:9]: Ditto on `f`.

File: sqlmesh/migrations/v0004_environmnent_add_finalized_at.py [hunk 3]
```diff
@@ -4,7 +4,9 @@
 
 def migrate(state_sync):  # type: ignore
     engine_adapter = state_sync.engine_adapter
-    environments_table = f"{state_sync.schema}._environments"
+    environments_table = f"_environments"
```
Comment (izeigerman) [inline sqlmesh/migrations/v0004_environmnent_add_finalized_at.py:7]: Ditto here and further

File: sqlmesh/migrations/v0005_create_seed_table.py [hunk 4]
```diff
@@ -4,7 +4,9 @@
 
 def migrate(state_sync):  # type: ignore
     engine_adapter = state_sync.engine_adapter
-    seeds_table = f"{state_sync.schema}._seeds"
+    seeds_table = f"._seeds"
```
Comment (izeigerman) [inline sqlmesh/migrations/v0005_create_seed_table.py:7]: Trailing dot

File: sqlmesh/schedulers/airflow/util.py [hunk 5]
```diff
@@ -36,13 +38,23 @@
...
         connection = Connection.get_connection_from_secrets(SQLMESH_STATE_CONN_ID)
 
+        connection_config_dict = json.loads(connection.extra)
+        state_schema = connection_config_dict.get("state_schema", state_schema)
+        if "type" not in connection_config_dict:
+            logger.info(
+                "SQLMesh connection in Airflow did not have type defined. "
+                "Therefore using Airflow database connection"
+            )
+            raise AirflowException
```
Comment (izeigerman) [inline sqlmesh/schedulers/airflow/util.py:52]: Are we raising a type here? Shouldn't it be `AirflowException()`?

Comment (eakmanrq) [reply sqlmesh/schedulers/airflow/util.py:52]: I think this works as expected:
```
def main():
    try:
        print("here")
        raise AirflowException
        print("here2")
    except AirflowException:
        print("exception")
```
```
here
exception
```

**Review by izeigerman — APPROVED** [review]

**Pull request merged**